### ISSUE TITLE
Taking forever to compile and reload browser

### ISSUE THREAD

**Issue by aaronbalthaser** [bdkjones/CodeKit-1-Old#662]
This started happening a couple weeks ago. Codekit is taking about 30 seconds to finish after I save the file. Im compiling SASS. I have to wait each time to see my changes. Ive restarted code kit while holding down shift and created my project over and also uninstalled Codekit and reinstalled.

**Comment by bdkjones** [comment]
Quit the app. Then, relaunch it while holding down shift. This will reset everything.

**Comment by aaronbalthaser** [comment]
I did that already, made no difference

**Comment by bdkjones** [comment]
Does this project use Compass sprites?

**Comment by aaronbalthaser** [comment]
Yes it does

**Comment by bdkjones** [comment]
Yep. That's a known bug in Compass.

There _is_ an issue on the Compass GitHub page that describes why it happens and includes a fix that you can implement to stop it. Basically, you need to go into the Compass files within CodeKit's app bundle and edit one line of code that's faulty in one of Compass's source files. That will stop Compass from recompiling the sprites every time you save a file, unless the sprites have changed. 

The Compass team has known about this for months but Compass hasn't been updated in... well... forever. 

-Bryan

**Comment by aaronbalthaser** [comment]
Thank you, at the sake of sounding lazy can you please forward me the link
to the issue? Thanks again.

**Comment by bdkjones** [comment]
Nope, but there's another issue thread on the CodeKit page (may be in the closed section) that has a link. A search should turn it up.

**Comment by aaronbalthaser** [comment]
Hello bkjjones, thanks for the direction to the link. I found the link. You were the one that posted it. Can you please help me out with understanding what Michael is saying. The link is:

michaelgilley/compass@f87c47d

I also found the line of code in codekit > compass. I guess the language may be ruby which I have no experience with. Michael says "Instead use compass compile my_file.scss --force." This doesn't even look like code that can be replace over "parser.options[:force] = true". Maybe I am misunderstanding this. Can you clarify, because I have been stuck using compass app with live reload and honestly it's a hassle and doesn't work very well. It seems like compass compile my_file.scss --force is suppose to override compass compile somewhere, but where? I want to go back to using Codekit. Thanks.

**Comment by aaronbalthaser** [comment]
Hello again, well it turns out that I am not using sprites in this project. While I normally use them in most projects I found I did not when I went to disable them in this project. Do you have any other suggestions that can determine this issue. Are there logs or any avenues I can pursue to help diagnose this issue?

**Comment by bdkjones** [comment]
Hi Aaron,

```
Please zip the entire project, send it to me, and tell me which file is taking a long time to compile so I can test on my end. Thanks.
```

**Comment by aaronbalthaser** [comment]
Hello, can you please give me the email address to where I should send the zip file. I made a couple attempts to send this via reply in my gmail and your server is rejecting the email. Excuse me for that, I am fairly new to this and have not yet built up the experience dealing with these scenarios. Thank you.

**Comment by bdkjones** [comment]
If your project is large, you'll need to upload it to something like Dropbox and then send me a link. Your email is probably being rejected because it's too big.

**Comment by aaronbalthaser** [comment]
Thank you for the tip. Here is the link. This is a Django project. I only included the media directory which contains the scss files. If you need more or anything else let me know please. Thank you.

https://www.dropbox.com/s/h5px5gk17ke2gak/media.zip

**Comment by aaronbalthaser** [comment]
Almost forgot, the issue was happening in all scss files. The _global.scss should yield the issue. Thanks.

**Comment by pietvanzoen** [comment]
I've been having a similar issue. For me it seems to be an issue with compiling a SVG filter in a LESS file. I've tried using LESS escaping but it still hangs. Here is the offending code that CodeKit doesn't seem to like:

`filter: url("data:image/svg+xml;utf8,<svg xmlns=\'http://www.w3.org/2000/svg\'><filter id=\'grayscale\'><feColorMatrix type=\'matrix\' values=\'0.3333 0.3333 0.3333 0 0 0.3333 0.3333 0.3333 0 0 0.3333 0.3333 0.3333 0 0 0 0 0 1 0\'/></filter></svg>#grayscale");`

CodeKit still takes about 5 seconds to compile with this line in the file, even if it's commented out. If I delete this line the compile is almost instant.

**Comment by pietvanzoen** [comment]
A little added info. This SVG filter in the same file doesn't seem to cause an issue:

`filter: url("data:image/svg+xml;utf8,<svg xmlns=\'http://www.w3.org/2000/svg\'><filter id=\'grayscale\'><feColorMatrix type=\'matrix\' values=\'1 0 0 0 0, 0 1 0 0 0, 0 0 1 0 0, 0 0 0 1 0\'/></filter></svg>#grayscale");`

**Comment by bdkjones** [comment]
What results do you get if you attempt to compile the file directly from the command line using Less.js?

**Comment by pietvanzoen** [comment]
Hmmm... yes it appears to do the same thing via command line. A LESS bug I guess.

**Comment by bdkjones** [comment]
Yep.

**Comment by aaronbalthaser** [comment]
Hello bdkjones, how are you doing? I was wondering if this issue was a bug with codekit or was it something else?

**Comment by bdkjones** [comment]
How fast does the project compile if you use the command line instead of CodeKit? If I recall, your project loads like 18,000 fonts and has a bunch of included frameworks. All of that is going to significantly slow down the Compass compiler, which is written in Ruby and is already dog-slow.

**Comment by aaronbalthaser** [comment]
Thanks, I guess I should have payed more attention to the docs on your site. I was using codekit without really understanding what is was doing behind the scenes. I think at some point I added some fonts to an assets directory just for storage. Had no idea Codekit was watching that directory. I removed it and everything works fine. Is there a way to have Codekit ignore directories? Thanks again. Sorry about that also.

**Comment by bdkjones** [comment]
Yep, use the Skipped Items list in the preferences. Edit the list before adding your project to the app.
